### PR TITLE
fix comparisons involving numbers

### LINKED ISSUE #13081: Some comparisons between rational and irrational numbers are incorrect
If you choose just the right rational number, you can end up in a situation where it is neither less than pi, nor equal to it, nor is pi less than it. This is with sympy 1.1.1, using Python 3.6.0 from Anaconda on Ubuntu 16.04.
```
>>> import sympy
>>> sympy.__version__
'1.1.1'
>>> r = sympy.Rational('905502432259640373/288230376151711744')
>>> r < sympy.pi
False
>>> r == sympy.pi
False
>>> sympy.pi < r
False
```
Of course, that same number is greater than pi, even though pi is not less than it.
```
>>> r > sympy.pi
True
```
I believe this is a result of using evalf() to do comparisons between rationals and reals... As we can see, this particular fraction happens to be exactly equal to pi if we use the default evalf precision of 15, but not if we use more.
```
>>> r == sympy.pi.evalf(15)
True
>>> r == sympy.pi.evalf(16)
False
```
Hopefully this isn't a duplicate issue; I did a bit of searching for related ones, and found the likes of #12583 and #12534. I think this is different than #12583 because I'm only concerned about comparisons where one of the numbers is rational. That should always be decidable - or am I misunderstanding something about math?

### PR DESCRIPTION
**Pull request by smichr**
fixes #13081 

cf #12583, #12534, #12537, 

- [x] add tests

### CODE REVIEW

**Comment by smichr** [comment]
Somewhere, a note about precision limits governing whether a certain relationship will evaluate or not might be good (like `pi < pi.n(300)` or something similar that you have cited, @asmeurer & @jksuom). Otherwise, do we go until precision/memory is exhausted to try prove something whenever we get an expression that doesn't evaluate with enough precision? I'm not sure how often these expressions are encountered and how much this will slow things down at the benefit of occasionally resolving some inequalities.

This is ready for a review, @asmeurer some time this week.

**Comment by asmeurer** [comment]
Does this address Float < Rational? I remember that being an issue that was discussed, but I don't see it in the tests you added.

**Comment by asmeurer** [comment]
c.f. https://github.com/sympy/sympy/pull/12537#issuecomment-297156228

File: sympy/core/tests/test_relational.py [hunk 1]
```diff
@@ -718,6 +718,30 @@ def test_issue_10927():
...
+    # 13081
+    r = Rational('905502432259640373/288230376151711744')
+    assert (r < pi) is S.false
+    assert (r > pi) is S.true
+    # 12583
+    v = sqrt(2)
+    u = sqrt(v) + 2/sqrt(10 - 8/sqrt(2 - v) + 4*v*(1/sqrt(2 - v) - 1))
+    assert (u >= 0) is S.true
+    # 12534
+    G = [Rational(pi.n(i)) > pi for i in (100, 101)]
```
Comment (asmeurer) [inline sympy/core/tests/test_relational.py]: Maybe some comments here explaining the digits of pi near 100 and how the rounding works there could be helpful.

**Comment by smichr** [comment]
> Does this address Float < Rational?

see lines 1293 ff. This is already traversed with tests. Do you have a pointer for a particular test that failed? (This wasn't my focus in this PR, but it is now currently handling the Float/Rational comparison as you and @jksuom discussed.)

File: sympy/core/numbers.py [hunk 2]
```diff
@@ -1288,11 +1288,14 @@ def __gt__(self, other):
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
-        if isinstance(other, NumberSymbol):
+        if other.is_NumberSymbol:
             return other.__lt__(self)
-        if other.is_comparable:
+        if other.is_Rational and not other.is_Integer:
```
Comment (smichr) [inline sympy/core/numbers.py]: <self is Float here>

**Comment by smichr** [comment]
> Does this address Float < Rational?

OK, here's a difference between this and master:

```
# in master
>>> any(i for i in range(15,50) if Rational(pi.n(i)) < pi.n(i))
False
>>> any(i for i in range(15,50) if pi.n(i)>Rational(pi.n(i)))
True
>>> all(i for i in range(15,50) if pi.n(i)>Rational(pi.n(i)))
True

# in this branch
>>> any(i for i in range(15,50) if Rational(pi.n(i)) < pi.n(i))
False
>>> any(i for i in range(15,50) if pi.n(i)>Rational(pi.n(i)))
False
```

**Comment by asmeurer** [comment]
Great that does look like an improvement. I might try and see if I can figure out how to reproduce some problems I was having in my other code that uses SymPy and if this branch fixes them.

**Comment by smichr** [comment]
> Great that does look like an improvement.

I added tests and annotations.

File: sympy/core/numbers.py (hunk 2 above)
Comment (ylemkimon) [reply sympy/core/numbers.py]: I think assigments to self should be avoided.

**Pull request merged**